src/Components/Address/useAddressAutocomplete.ts

```diff
@@ -75,6 +75,15 @@ export const useAddressAutocomplete = (
     [apiKey]
   )
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const throttledFetchSuggestions = useCallback(
+    throttle(fetchSuggestions, 2000, {
+      leading: true,
+      trailing: true,
+    }),
+    [fetchSuggestions, throttle]
```

Comment:
non-blocking: this should technically be fine, but `[]` might be even safer. We only want it to execute once, and you have to look at this code to convince yourself it will do that.

Reply:
I'm not sure what you mean here, can you provide more context?

Reply:
are you saying to use an empty array instead of deps?